src/cryptography/x509.py

```diff
@@ -237,6 +237,40 @@ def __repr__(self):
                 "path_length={0.path_length})>").format(self)
 
 
+class KeyUsage(object):
+    def __init__(self, digital_signature, non_repudiation, key_encipherment,
+                 data_encipherment, key_agreement, key_cert_sign, crl_sign,
+                 encipher_only, decipher_only):
+        if not key_agreement and (encipher_only or decipher_only):
+            raise ValueError(
+                "encipher_only and decipher_only can only be true when "
+                "key_agreement is true"
+            )
+
+        self._digital_signature = digital_signature
+        self._non_repudiation = non_repudiation
+        self._key_encipherment = key_encipherment
+        self._data_encipherment = data_encipherment
+        self._key_agreement = key_agreement
+        self._key_cert_sign = key_cert_sign
+        self._crl_sign = crl_sign
+        self._encipher_only = encipher_only
+        self._decipher_only = decipher_only
+
+    digital_signature = utils.read_only_property("_digital_signature")
+    non_repudiation = utils.read_only_property("_non_repudiation")
+    # Later versions of X509 have renamed non_repudiation to
+    # content_commitment
+    content_commitment = utils.read_only_property("_non_repudiation")
+    key_encipherment = utils.read_only_property("_key_encipherment")
+    data_encipherment = utils.read_only_property("_data_encipherment")
+    key_agreement = utils.read_only_property("_key_agreement")
+    key_cert_sign = utils.read_only_property("_key_cert_sign")
+    crl_sign = utils.read_only_property("_crl_sign")
+    encipher_only = utils.read_only_property("_encipher_only")
+    decipher_only = utils.read_only_property("_decipher_only")
```

Comment:
Would you prefer it to raise an error if accessed or something? The statement that this value is "undefined" when `key_agreement` is false in the RFC is really annoying...